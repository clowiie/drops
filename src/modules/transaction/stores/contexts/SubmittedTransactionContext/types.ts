export interface ContextType {
  submittedHash: string
  setSubmittedHash: (hash: string) => void
  clearSubmitted: () => void
}
