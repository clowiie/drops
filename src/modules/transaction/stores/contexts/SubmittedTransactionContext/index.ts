import { createContext } from 'react'

import { ContextType } from './types'

const SubmittedTransactionContext = createContext<ContextType>({
  submittedHash: '',
  setSubmittedHash: () => {},
  clearSubmitted: () => {},
})

export default SubmittedTransactionContext
