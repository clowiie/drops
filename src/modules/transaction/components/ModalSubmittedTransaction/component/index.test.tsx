import React from 'react'

import { render } from '@testing-library/react'

import combineWrappers from 'test/utils/combineWrappers'
import createContextWrapper from 'test/utils/createContextWrapper'
import createPortalWrapper from 'test/utils/createPortalWrapper'
import mockComponent from 'test/utils/mockComponent'

import SubmittedTransactionContext from 'modules/transaction/stores/contexts/SubmittedTransactionContext'

import * as ModalSubmittedTransactionType from '.'

describe('ModalSubmittedTransaction', () => {
  const ModalSpy = mockComponent('common/components/Modal')

  const clearSubmittedSpy = jest.fn()

  const { default: ModalSubmittedTransaction } =
    require('.') as typeof ModalSubmittedTransactionType

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should match snapshots', () => {
      const { container, getByText } = render(
        <ModalSubmittedTransaction isOpen />,
        {
          wrapper: combineWrappers(
            createContextWrapper(SubmittedTransactionContext, {
              submittedHash: 'hash',
              setSubmittedHash: () => {},
              clearSubmitted: clearSubmittedSpy,
            }),
            createPortalWrapper(),
          ),
        },
      )

      expect(container.firstChild).toMatchSnapshot()
      expect(getByText('Transaction Submitted')).toBeDefined()
      expect(getByText('View transaction')).toBeDefined()
      expect(getByText('View transaction').hasAttribute('href')).toBe(true)
      expect(ModalSpy).toBeCalledTimes(1)
    })

    it('should match snapshots with isOpen is false', () => {
      const { container } = render(
        <ModalSubmittedTransaction isOpen={false} />,
        {
          wrapper: combineWrappers(
            createContextWrapper(SubmittedTransactionContext, {
              submittedHash: 'hash',
              setSubmittedHash: () => {},
              clearSubmitted: clearSubmittedSpy,
            }),
            createPortalWrapper(),
          ),
        },
      )

      expect(container.firstChild).toMatchSnapshot()
      expect(ModalSpy).toBeCalledTimes(1)
    })
  })
})
