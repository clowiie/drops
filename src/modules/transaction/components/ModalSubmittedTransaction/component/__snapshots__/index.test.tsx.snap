// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSubmittedTransaction snapshots should match snapshots 1`] = `
<div
  id="root"
/>
`;

exports[`ModalSubmittedTransaction snapshots should match snapshots with isOpen is false 1`] = `
<div
  id="root"
/>
`;
